core/src/broadcast_stage/slot_transmit_shreds_cache.rs

```diff
@@ -0,0 +1,675 @@
+use super::*;
+use solana_ledger::shred::Shred;
+use solana_sdk::{clock::Slot, pubkey::Pubkey};
+use std::collections::VecDeque;
+
+pub type TransmitShreds = (Option<Arc<HashMap<Pubkey, u64>>>, Arc<Vec<Shred>>);
+
+#[derive(Default, Debug, PartialEq)]
+pub struct SlotCachedTransmitShreds {
+    pub slot: Slot,
```

Comment:
Do we need slot here? the cache already maps slot to this cache entry, right?

Reply:
hehe I actually specifically added it just for the `update_from_blockstore()` so I wouldn't have to pass the `slot` into that function/risk somebody passing in the wrong slot. What do you think, should the `slot` just be passed in then for that function?